### PR TITLE
fix: corrections for wasm tls

### PR DESCRIPTION
**Pull request by haraldh**
How to test:

```
$ git clone https://github.com/haraldh/mini_http
$ cd mini_http
$ cargo +nightly build --target wasm32-wasi --example hello_enarx
$ <PATH_TO_THIS_PRS_ENARX_BIN> run \
  --wasmcfgfile examples/Enarx.toml \
  target/wasm32-wasi/debug/examples/hello_enarx.wasm
```

Then point your browser to [`https://127.0.0.1:3000`](https://127.0.0.1:3000)

### CODE REVIEW

**Comment by rjzak** [comment]
I find that this works with Firefox and curl, but I seem to have to run `cargo clean` and re-build a few times for changes to be run. Is this along the lines of what Ben was talking about a few days ago? I don't know if this is an issue or not, but curl has this message: `ALPN, server did not agree to a protocol`, seems to relate to an HTTP2 & TLS https://serverfault.com/questions/920928/server-certificate-verification-ok-but-alpn-server-did-not-agree-to-a-protoc

File: src/bin/wasmldr/src/loader/compiled/tls.rs [hunk 1]
```diff
@@ -163,7 +171,7 @@ impl WasiFile for Stream {
     }
 
     async fn read_vectored<'a>(&self, bufs: &mut [IoSliceMut<'a>]) -> Result<u64, Error> {
-        self.complete_io()?;
+        self.complete_io(TlsComplete::Read)?;
```
[the remark this answers is not in the document]
Comment (haraldh) [reply src/bin/wasmldr/src/loader/compiled/tls.rs]: done

File: src/bin/wasmldr/src/loader/compiled/tls.rs [hunk 2]
```diff
@@ -11,18 +12,82 @@ use cap_std::net::{TcpListener as CapListener, TcpStream as CapStream};
...
+
+trait IOAsync {
+    fn complete_io_async<T>(&mut self, _io: &mut T) -> Result<(usize, usize), io::Error>
+    where
+        Self: Sized,
+        T: io::Read + io::Write;
+}
+
+impl IOAsync for Connection {
+    fn complete_io_async<T>(&mut self, io: &mut T) -> Result<(usize, usize), io::Error>
```
[the remark this answers is not in the document]
Comment (haraldh) [reply src/bin/wasmldr/src/loader/compiled/tls.rs:43]: done

**Review by rvolosatovs — DISMISSED** [review]

File: src/bin/wasmldr/src/loader/compiled/tls.rs [hunk 3]
```diff
@@ -11,18 +12,87 @@ use cap_std::net::{TcpListener as CapListener, TcpStream as CapStream};
...
         InvalidInput => ErrorKind::Inval.into(),
         Unsupported => ErrorKind::Notsup.into(),
-        _ => error.into(),
+        InvalidData => ErrorKind::Inval.into(),
+        _ => Error::from(ErrorKind::Io).context(error),
+    }
+}
+
+trait IOAsync {
+    fn complete_io_async<T>(&mut self, _io: &mut T) -> Result<(usize, usize), io::Error>
```
Comment (rvolosatovs) [inline src/bin/wasmldr/src/loader/compiled/tls.rs:31]:
```suggestion
    fn complete_io_async<T>(&mut self, io: &mut T) -> Result<(usize, usize), io::Error>
```

Comment (haraldh) [reply src/bin/wasmldr/src/loader/compiled/tls.rs]: fixed

**Review by rvolosatovs — APPROVED** [review]
Thanks!

**Pull request merged**